### DISCUSSION TITLE
Getting error while importing Sionna (incompatible Mitsuba version)

### DISCUSSION THREAD

**Discussion by bnss0720** [NVlabs/sionna#682] [General]
Hi,

I am using google collab for working on Sionna examples and development.

Till last week everything was working fine, but suddenly from today I am getting the following import error:

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
[<ipython-input-1-a02726b8b86b>](https://localhost:8080/#) in <cell line: 9>()
      9 try:
---> 10     import sionna as sn
     11 except ImportError as e:

ModuleNotFoundError: No module named 'sionna'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
5 frames
[/usr/local/lib/python3.10/dist-packages/sionna/rt/camera.py](https://localhost:8080/#) in Camera()
     51     # convention.
     52     # Note: Mitsuba uses degrees
**---> 53     mi_2_sionna = ( mi.ScalarTransform4f.rotate([0,0,1], 90.0)**
     54                 @ mi.ScalarTransform4f.rotate([1,0,0], 90.0) )
     55 

**TypeError: rotate(): incompatible function arguments. The following argument types are supported:
    1. rotate(self, axis: mitsuba.ScalarPoint3f, angle: float) -> mitsuba.ScalarTransform4f**

Invoked with types: list, float
```
---------------------------------------------------------------------------

Any idea what might have gone wrong?

Note : the pip install sionna is fine from the below code snippet.

```py
try:
    import sionna as sn
except ImportError as e:
    # Install Sionna if package is not already installed
    import os
    **os.system("pip install sionna")**
    import sionna as sn
```

Thanks

**Comment by SebastianCa** [comment]
Hi @bnss0720,

This issue is due to an incompatibility with `mitsuba==3.6`. It will be fixed in the next release of Sionna.
As a workaround, you can try manually downgrading to `mitsuba==3.5`.

**Reply by bnss0720** [reply]
Thanks for the response, I can see that mitsuba is 3.6 on myside. Could you please help where exactly and how should I downgrade to mitsuba==3.5. I tried several ways, but getting errors, not sure whether I am doing correctly or not. Not sure do I have to do it before or after sionna import. Either way I am getting error.

```py
try:
    import sionna as sn
except ImportError as e:
    # Install Sionna if package is not already installed
    import os
    **os.system("pip install sionna")**
    import sionna as sn
```

**Reply by SebastianCa** [reply]
Have you tried `pip install --upgrade mitsuba==3.5.2`?

**Reply by bnss0720** [reply]
yeah it works now. 
Post downgrading to mitsuba==3.5.2, I did 'import mitsuba ' , which I shouldn't have done. 
When I do import mitsuba , it asks for mitsuba variant. 
So I am not doing any import of mitsuba , just downgraded to mitsuba==3.5.2 and it works now.

Thanks for the support.
